File: bst.py
""" Node is defined as

class node:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None
"""


def checkBST(root):
    return checkBSTfunc(root,1,10000)

def checkBSTfunc(root,min_num,max_num):
    
    if root == None:
        return True
    
    if root.data > max_num or root.data < min_num:
        return False
    
    else:
        check1 = checkBSTfunc(root.left,min_num,root.data-1)
        check2 = checkBSTfunc(root.right,root.data+1,max_num)
        return (check1 and check2)

File: bst.cpp
 bool checkBST(Node* root, int min, int max) {

       if(root==NULL){
           return true;
       }

       if(root->data > max || root->data < min){
           return false;
       }

      return ( checkBST(root->left, min, root->data-1) && checkBST(root->right, root->data+1, max) );

}


bool checkBST(Node* root) {

    return checkBST(root,0,10000);
}
